atomicfu/build.gradle.kts

```
@@ -0,0 +1,382 @@
+    dependsOn(tasks.get("compileTestKotlinJvm"))
+    mainClass = "kotlinx.atomicfu.transformer.AtomicFUTransformerKt"
+    args(classesPreAtomicFuDir, classesPostTransformFU, "FU")
+    classpath(transformer)
+    inputs.dir(classesPreAtomicFuDir)
+    outputs.dir(classesPostTransformFU)
+}
+
+val transformBOTH by tasks.registering(JavaExec::class) {
+    dependsOn("compileTestKotlinJvm")
+    mainClass = "kotlinx.atomicfu.transformer.AtomicFUTransformerKt"
+    args(classesPreAtomicFuDir, classesPostTransformBOTH, "BOTH")
+    classpath = transformer
+    inputs.dir(classesPreAtomicFuDir)
+    outputs.dir(classesPostTransformBOTH)
+}
+
+
+val transformVH by tasks.registering(JavaExec::class) {
+    dependsOn("compileTestKotlinJvm")
+    mainClass = "kotlinx.atomicfu.transformer.AtomicFUTransformerKt"
+    args(classesPreAtomicFuDir, classesPostTransformVH, "VH")
+    classpath(transformer)
+    inputs.dir(classesPreAtomicFuDir)
+    outputs.dir(classesPostTransformVH)
+}
+
+
+val transformedTestFU_current by tasks.registering(Test::class) {
+    dependsOn(transformFU)
+    classpath = files(configurations.getByName("jvmTestRuntimeClasspath"), classesPostTransformFU)
+    testClassesDirs = project.files(classesPostTransformFU)
+    exclude("**/*LFTest.*", "**/TraceToStringTest.*", "**/AtomicfuReferenceJsTest.*")
+    filter { isFailOnNoMatchingTests = false }
+}
+
+val transformedTestBOTH_current by tasks.registering(Test::class) {
+    dependsOn(transformBOTH)
+    classpath = files(configurations.getByName("jvmTestRuntimeClasspath"), classesPostTransformBOTH)
+    testClassesDirs = project.files(classesPostTransformBOTH)
+    exclude(
+        "**/*LFTest.*",
+        "**/TraceToStringTest.*",
+        "**/TopLevelGeneratedDeclarationsReflectionTest.*",
+        "**/SyntheticFUFieldsTest.*",
+        "**/AtomicfuReferenceJsTest.*"
+    )
+    filter { isFailOnNoMatchingTests = false }
+}
+
+val transformedTestVH by tasks.registering(Test::class) {
+    dependsOn(transformVH)
+    classpath = files(configurations.getByName("jvmTestRuntimeClasspath"), classesPostTransformVH)
+    testClassesDirs = project.files(classesPostTransformVH)
+    exclude(
+        "**/*LFTest.*",
+        "**/TraceToStringTest.*",
+        "**/TopLevelGeneratedDeclarationsReflectionTest.*",
+        "**/SyntheticFUFieldsTest.*",
+        "**/AtomicfuReferenceJsTest.*"
+    )
+    filter { isFailOnNoMatchingTests = false }
+
+    onlyIf { currentTask ->
+        logger.info("Current java version for task ${currentTask.name} is : ${JavaVersion.current()}")
+        JavaVersion.current().ordinal >= JavaVersion.VERSION_1_9.ordinal
+    }
+}
+val jvmTestAll by tasks.registering {
+    dependsOn(
+        transformedTestFU_current,
+        transformedTestBOTH_current,
+        transformedTestVH
+    )
+}
+
+tasks.withType(Test::class.java) {
+    testLogging {
+        showStandardStreams = true
+        events("passed", "failed")
+    }
+}
```

```suggestion
tasks.withType<Test>().configureEach {
    testLogging {
        showStandardStreams = true
        events("passed", "failed")
    }
}
```

Suggestion: use Kotlin DSL helper functions to get all Test tasks, and lazily configure each task.